src/compiler/program.ts

```diff
@@ -972,21 +959,93 @@ namespace ts {
         return resolutions;
     }
 
-    export function getDefaultTypeDirectiveNames(options: CompilerOptions, rootFiles: string[], host: CompilerHost): string[] {
+
+    /**
+      * Computes the types folder, a.k.a. the primary types lookup location.
+      * This is commonly './types' relative to whatever the compilation root is.
+      */
+    function getTypesFolderLocation(options: CompilerOptions, host: CompilerHost, inferredRoot: string): Path {
+        // Openish question: When tsconfig specifies a rootDir *and* a relative typesRoot, is the typesRoot
+        // relative to the tsconfig file path, or the rootDir path?

```

Comment:
I started out thinking that it should be relative to the `rootDir`, but I think that if someone has an *explicit* relative `typesRoot`, then that should be relative to the `tsconfig.json`.

A user should be able to open up a tsconfig and know the `typesRoot` without doing any extra calculations.